nifi-nar-bundles/nifi-aws-bundle/nifi-aws-parameter-value-providers/src/main/java/org/apache/nifi/stateless/parameter/aws/SecretsManagerParameterValueProvider.java

```diff
@@ -0,0 +1,172 @@
+ *      nifi.stateless.parameter.provider.AWSSecretsManager.properties.aws-credentials-file=./conf/bootstrap-aws.conf
+ * </code>
+ */
+public class SecretsManagerParameterValueProvider extends AbstractParameterValueProvider implements ParameterValueProvider {
+    private static final String QUALIFIED_SECRET_FORMAT = "%s/%s";
+    private static final String ACCESS_KEY_PROPS_NAME = "aws.access.key.id";
+    private static final String SECRET_KEY_PROPS_NAME = "aws.secret.access.key";
+    private static final String REGION_KEY_PROPS_NAME = "aws.region";
+
+    public static final PropertyDescriptor AWS_CREDENTIALS_FILE = new PropertyDescriptor.Builder()
+            .displayName("AWS Credentials File")
+            .name("aws-credentials-file")
+            .required(false)
+            .defaultValue("./conf/bootstrap-aws.conf")
+            .description("Location of the bootstrap-aws.conf file that configures the AWS credentials.  If not provided, the default AWS credentials will be used.")
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .build();
+
+    private final Set<String> supportedParameterNames = new HashSet<>();
+
+    private List<PropertyDescriptor> descriptors;
+
+    private AWSSecretsManager secretsManager;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @Override
+    protected void init(final ParameterValueProviderInitializationContext context) {
+        super.init(context);
+
+        this.descriptors = Collections.singletonList(AWS_CREDENTIALS_FILE);
+
+        final String awsCredentialsFilename = context.getProperty(AWS_CREDENTIALS_FILE).getValue();
+        try {
+            this.secretsManager = this.configureClient(awsCredentialsFilename);
+        } catch (final IOException e) {
+            throw new IllegalStateException("Could not configure AWS Secrets Manager Client", e);
+        }
+
+        cacheSupportedParameterNames();
+    }
+
+    @Override
+    public boolean isParameterDefined(final String contextName, final String parameterName) {
+        return supportedParameterNames.contains(getSecretName(contextName, parameterName));
+    }
+
+    @Override
+    public String getParameterValue(final String contextName, final String parameterName) {
+        final String secretName = getSecretName(contextName, parameterName);
+        final GetSecretValueRequest getSecretValueRequest = new GetSecretValueRequest()
+                .withSecretId(secretName);
+        final GetSecretValueResult getSecretValueResult = secretsManager.getSecretValue(getSecretValueRequest);
+
+        if (getSecretValueResult == null) {
+            throw new IllegalArgumentException(String.format("Secret [%s] not found", secretName));
+        }
+        if (getSecretValueResult.getSecretString() != null) {
+            return getSecretValueResult.getSecretString();
```

Comment:
I added a secret in SecretManager named MyContext/MyParameter. I used an "Other type of secret" and I set the key to "MyContext/MyParameter" and the value to "Hello". I expected the parameter to resolve to `Hello`. Instead, it resolved to `{"MyContext/MyParameter":"Hello"}`. Am I using this wrong?

Reply:
Are you adding the secret as a plain text secret?  If you use the AWS key/value interface, it creates the JSON you posted instead of simply "Hello".  Basically, if you use the command line example in the PR, it should resolve to "Hello".  I will also take a look at this again, in case it is a bug in the code, though.

Reply:
Ahhh yes. I used the key/value interface. The README gives an example of using the CLI, but we should probably indicate in the README also that if you're going to use the AWS Console (i.e., the UI) you need to create the secret and use a type of "Other type of secrets (e.g. API key)" and then go to the plaintext tab, remove the JSON completely, and then just type the secret value that you'd like to use.